bundle/src/main/java/com/adobe/acs/samples/models/SomeApiModel.java

```diff
@@ -0,0 +1,24 @@
+package com.adobe.acs.samples.models;
+
+import com.adobe.acs.samples.models.api.SomeApi;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.Self;
+
+import javax.inject.Inject;
+
+@Model(adaptables = ResourceResolver.class, adapters = SomeApi.class)
```

Comment:
Would be good to explain when/why you'd want to do this over the usual straight adaptables to model impl pattern, and what happens if there are multiple impls of the same interface.